BLSQ/iaso
# IA-2813 allow event date to come from the period

Currently iaso use the created_at of the submission to export dhis2 event.
Some project want that we use the submission period (in this special case the daily period)


Related JIRA tickets : IA-2813

## Self proofreading checklist

- [ ] Are my typescript files well typed (existing code wasn't typed)
- [x] Are there enough tests (only python)

## Doc

- no doc

## Changes

Since we have existing data collection that assume that the created_at is used (what ever there's a period to the form/submission)

- this adds a field `event_date_source` on the form MappingVersion
   - if can be FROM_SUBMISSION_CREATED_AT or FROM_SUBMISSION_PERIOD 
- this also add a start_date() on the periods (python side) so we can turn the dhis2 period
- make this editable in the form mapping version "tree" screen
- the code exporting these data to dhis2 will use that `event_date_source` if none it will default the created_at behavior

## How to test

1. `docker-compose run --rm iaso manage seed_test_data --mode=seed --dhis2version=2.39`
2. go in the forms list -> Community Verification Satisfaction form -> mapping icon
3. open the mapping
4. then change the 'Event date source' for use the period
![image](https://github.com/BLSQ/iaso/assets/371692/245839ec-a564-4df5-af8a-f2cf2087b5b5)

6. export to dhis2
   - pick one of the submission export it 
   - verify https://play.dhis2.org/2.39.4.1/api/events/EN2LtXfodEb.json (EN2LtXfodEb = the export_id in instance)
   - you can modify the "assigned" period in the instance details screen
  
![image](https://github.com/BLSQ/iaso/assets/371692/6f48e041-3985-4585-a543-e6b89eb62f27)

   - export again and verify the eventDate has been updated

![image](https://github.com/BLSQ/iaso/assets/371692/a41f576d-a6aa-4ee0-aa63-3b03e1ebac99)



## Print screen / video

![image](https://github.com/BLSQ/iaso/assets/371692/245839ec-a564-4df5-af8a-f2cf2087b5b5)

## Notes

the periods to "date" might need to be reviewed if start supporting the Ethiopian calendar
